cpp/oneapi/dal/algo/triangle_counting/backend/cpu/vertex_ranking_default_kernel_scalar.hpp

```diff
@@ -68,37 +70,27 @@ ONEDAL_FORCEINLINE std::int64_t intersection_local_tc(const Index* neigh_u,
 
 template <typename Cpu>
 ONEDAL_FORCEINLINE array<std::int64_t> triangle_counting_local_(
-    const dal::preview::detail::topology<std::int32_t>& data,
+    const dal::preview::detail::topology<std::int32_t>& t,
     int64_t* triangles_local) {
-    const auto g_edge_offsets = data._rows.get_data();
-    const auto g_vertex_neighbors = data._cols.get_data();
-    const auto g_degrees = data._degrees.get_data();
-    const auto g_vertex_count = data._vertex_count;
-    const auto g_edge_count = data._edge_count;
-    std::int32_t average_degree = g_edge_count / g_vertex_count;
-    int thread_cnt = dal::detail::threader_get_max_threads();
+    const auto vertex_count = get_topology_vertex_count(t);
+    std::int32_t average_degree = get_topology_edge_count(t) / vertex_count;
+    int thread_count = dal::detail::threader_get_max_threads();
 
-    dal::detail::threader_for(thread_cnt * g_vertex_count,
-                              thread_cnt * g_vertex_count,
+    dal::detail::threader_for(thread_count * vertex_count,
+                              thread_count * vertex_count,
                               [&](std::int64_t u) {
                                   triangles_local[u] = 0;
                               });
 
     const std::int32_t average_degree_sparsity_boundary = 4;
     if (average_degree < average_degree_sparsity_boundary) {
-        dal::detail::threader_for(g_vertex_count, g_vertex_count, [&](std::int32_t u) {
-            for (auto v_ = g_vertex_neighbors + g_edge_offsets[u];
-                 v_ != g_vertex_neighbors + g_edge_offsets[u + 1];
-                 ++v_) {
-                std::int32_t v = *v_;
+        dal::detail::threader_for(vertex_count, vertex_count, [&](std::int32_t u) {
+            for (auto v: get_topology_vertex_neighbors(t, u)) {
                 if (v > u) {
                     break;
                 }
-                auto u_neighbors_ptr = g_vertex_neighbors + g_edge_offsets[u];
-                for (auto w_ = g_vertex_neighbors + g_edge_offsets[v];
-                     v_ != g_vertex_neighbors + g_edge_offsets[v + 1];
-                     ++w_) {
-                    std::int32_t w = *w_;
+                auto u_neighbors_ptr = get_topology_vertex_neighbors_begin(t, u);
+                for (auto w: get_topology_vertex_neighbors(t, v)) {
                     if (w > v) {
```

Comment:
mostly opt topic, not relevant to style - can we iterate till v or so?

Reply:
I see no ways here. The place of v is unpredictable as graph is in CSR format.